### PR TITLE
Add endpoints to proxy-config all output

### PR DESCRIPTION
**Pull request by keithmattix**
**Please provide a description of this PR:**
Fix the output of `istioctl proxy-config all` to include EDS configuration when the `--json` or `--yaml` flags are used. Also add a test

### CODE REVIEW

**Review by howardjohn — APPROVED** [review]

File: releasenotes/notes/fix-proxyconfig-endpoints-json.yaml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+apiVersion: release-notes/v2
+kind: bug-fix
```
Comment (hzxuzhonghu) [inline releasenotes/notes/fix-proxyconfig-endpoints-json.yaml:2]: nit: this is a enhancement

Comment (keithmattix) [reply releasenotes/notes/fix-proxyconfig-endpoints-json.yaml:2]: Is it? `istioctl proxyconfig all` included endpoints as long as the output wasn't YAML or JSON. That sounds like a bug to me

Comment (hanxiaop) [reply releasenotes/notes/fix-proxyconfig-endpoints-json.yaml:2]: IIRC endpoints summary was added by me, maybe I forgot handling the yaml/json format. It should be a bug.

Comment (hanxiaop) [reply releasenotes/notes/fix-proxyconfig-endpoints-json.yaml:2]: Thanks for the fix

**Review by hanxiaop — APPROVED** [review]

**Pull request merged**